### PR TITLE
Refactor Reset Form & minor fixes

### PR DESCRIPTION
**Pull request by masskoder**
### Scope of changes

This PR brings several fixes to improve the user experience:

 - clear data has been added, allowing users to easily reset the form and clear all inputted data.
- An Unsaved modal now appears for each form stepper, alerting users of any unsaved changes before navigating away.
- The delete section has been improved and is now functioning correctly,
 
### Type of change

- [x] bug fix
- [ ] new feature
- [ ] documentation
- [x] refactor

### Author checklist

- [ ] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [x]   Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.

### CODE REVIEW

File: web/gds-user-ui/src/components/RegistrationForm/InvalidFormPrompt.tsx
Comment (daniellemaxwell) [inline web/gds-user-ui/src/components/RegistrationForm/InvalidFormPrompt.tsx:49]: During QA, Edwin stated the text could state both button options and to handle this change in the future.

**Review by daniellemaxwell — COMMENTED** [review]
This is a good start, but two issues blocking the release do not appear to be fixed: 
- When the section is reset, the user must refresh the page for the data to no longer appear
- When the form is reset, the user must refresh for all of the data in each individual form to no longer appear

Please see this video for more details: https://www.awesomescreenshot.com/video/17946941?key=055205a3a8146957c022ac8a3d6ad4fe

**Comment by daniellemaxwell** [comment]
@masskoder Also, when a user imports a JSON file, they must refresh the page to see the data.

File: web/gds-user-ui/src/hooks/useUploadFile.tsx
Comment (daniellemaxwell) [reply web/gds-user-ui/src/hooks/useUploadFile.tsx:17]: The error message could be a little more user friendly, but that's not an urgent thing and could be handled in a future sprint.
Screenshot 2023-06-01 at 3 07 24 PM

By the way, I have noticed that if a user imports an invalid file and then attempts to import another invalid file, they do not see the error message the 2nd time.

File: web/gds-user-ui/src/components/BasicDetail/index.tsx
Comment (masskoder) [reply web/gds-user-ui/src/components/BasicDetail/index.tsx:91]: we do not need this check for the others, only the basic step need this check because when the user reset all the form, he is redirected to the basic form.
For the other form, I just check if that step is deleted from the state.

File: web/gds-user-ui/src/hooks/useUploadFile.tsx
Comment (masskoder) [reply web/gds-user-ui/src/hooks/useUploadFile.tsx:17]: Thanks just noticed that. it's going to be a future fix.

File: web/gds-user-ui/src/components/RegistrationForm/InvalidFormPrompt.tsx
Comment (masskoder) [reply web/gds-user-ui/src/components/RegistrationForm/InvalidFormPrompt.tsx:49]: i make this update during that call and forgot to remove it. I think you have a PR that should display this part as you suggested.

Comment (daniellemaxwell) [reply web/gds-user-ui/src/components/RegistrationForm/InvalidFormPrompt.tsx:49]: I've merged that PR, so this should be removed.

**Review by daniellemaxwell — APPROVED** [review]
Adding the hard reload is working for when a user wants to clear a section or import a valid JSON file. This should be added after a user clicks to refresh the form as well to make sure that works properly as I still saw data when I was sent to the beginning of the form. Also, when an invalid JSON file is uploaded, a hard reload should occur so that a user will see an error if they attempt to submit another invalid JSON file. 

This should be good to merge once those things are handled!

**Pull request merged**
